plugins/parodos/src/components/workflow/workflowDetail/topology/useDemoPipelineNodes.tsx

```diff
@@ -22,7 +22,8 @@ export function useDemoPipelineNodes(
     if (status.toUpperCase() === 'COMPLETED') return RunStatus.Succeeded;
     else if (status.toUpperCase() === 'IN_PROGRESS')
       return RunStatus.InProgress;
-    else if (status.toUpperCase() === 'FAILED') return RunStatus.Failed;
+    else if (['FAILED', 'REJECTED'].indexOf(status.toUpperCase()) > -1)
```

Comment:
We can use `Array.includes()`

Reply:
Do we need `toUpperCase`, either the API returns uppercase statuses or we use zod to transform them to uppercase then we can use typescript for strict typing around the status.

You definitely should never need to do this and could be a source of bugs if we are not doing this at a higher level.

Reply:
A status should always be a union or part of a discriminated union which means it can only be from a restricted set of values.  modifying the string to check this breaks it.

Reply:
One more small nit,

I think not having braces around `if` is another thing that can catch you out

so 

```ts
if (status.toUpperCase() === 'COMPLETED') return RunStatus.Succeeded;
```

should be 

```ts
if (status.toUpperCase() === 'COMPLETED') {
  return RunStatus.Succeeded;
}
```

This is a personal thing and definitely not saying you need to do this.

We could use eslint and `--fix` and the [curly](https://eslint.org/docs/latest/rules/curly) rule to change the entire codebase but this is just a suggestion and IMO a source of silly bugs.

So if you don't think this is necessary then ignore.

Reply:
we don't have many tests if any so I'd like to lean on eslint and typescript more or.....add more tests

Reply:
Per my earlier comment - I agree, the API should return constants consistently.

Regarding braces: agreement as well. It's a good idea to leverage the `eslint` to enforce/fix that.

Reply:
@mareklibra I totally missed your comment, apologies.

Reply:
and totally agree :).